src/skmultiflow/data/base_generator.py

```diff
@@ -2,26 +2,26 @@
 from skmultiflow.core.base_object import BaseObject
 
 
-class Stream(BaseObject, metaclass=ABCMeta):
-    """ The abstract class setting up the minimum requirements of a stream,
+class BaseGenerator(BaseObject, metaclass=ABCMeta):
+    """ The abstract class setting up the minimum requirements of a generator,
     so that it can work along the other modules in the scikit-multiflow
     framework.
-    
+
     Raises
     ------
     NotImplementedError: This is an abstract class.
-    
+
     """
+    # TODO : remove everything related to features vs targets, this sould not be handled at this level
     def __init__(self):
         self.n_samples = 0
         self.n_targets = 0
         self.n_features = 0
         self.n_num_features = 0
         self.n_cat_features = 0
         self.n_classes = 0
-        self.cat_features_idx = []
-        self.current_sample_x = None
-        self.current_sample_y = None
+        self.cat_features_idx = []  # TODO : remove
```

Comment:
Won't remove given that `cat_features_idx` is used by some learning methods that implement specific ways to process categorical features.

Reply:
@jacobmontiel agreed for this PR I'll re-add this